Add TCP Echo Server Example

dave-tucker opened the pull request:
This example illustrates how to use socket redirection

Signed-off-by: Dave Tucker <dave@dtucker.co.uk>

tohojo commented:
Thanks for the example. But, erm, what does it do? :)

I tried parsing the BPF example, and I'm not actually sure how it works; could you please add a bit more description to the README explaining this? Also, for us non-rustaceans, some hints as to which bit is supposed to be BPF code and which bit is userspace code would be helpful...

netoptimizer commented:
I'm having a hard time finding the BPF code in this example.
I assume it is in echo/echo-ebpf/src/main.rs, but I simply don't understand this code.
We need some more hand-holding the reader that want to learn from this bpf-example, before this can get merged.

dave-tucker commented:
Ack. I'm working on addressing your comments and will push something soon. Hopefully later today! 

On Fri, 19 Nov 2021, at 09:52, Jesper Dangaard Brouer wrote:
> 
> 
> 
> I'm having a hard time finding the BPF code in this example.
> I assume it is in echo/echo-ebpf/src/main.rs, but I simply don't understand this code.
> We need some more hand-holding the reader that want to learn from this bpf-example, before this can get merged.
> 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/xdp-project/bpf-examples/pull/28#issuecomment-973918874>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA43JEJABILZQAQJALYZZ7TUMYM6RANCNFSM5IENGBEQ>.
>
